### ISSUE TITLE
[BUG] push_notebook doesn't work when LabelSet is only listener to a data source

### ISSUE THREAD

**Issue by currivan** [bokeh/bokeh#9294]
### FOLLOW THESE INSTRUCTIONS CAREFULLY

*ISSUES THAT DO NOT CONTAIN NECESSARY INFORMATION MAY BE CLOSED, IMMEDIATELY*

The issue tracker is NOT the place for general support. For questions and technical assistance, come join the [Bokeh  Project Discourse](https://discourse.bokeh.org).

For defects or deficiencies, please provide *ALL OF THE FOLLOWING*:

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
bokeh: 1.3.4
python: 3.6.8
jupyterlab: 0.35.4
macos: 10.14.5
chrom: 76.0.3809.100 

#### Description of expected behavior and the observed behavior

push_notebook() correctly updates text labels when there's a listener such as scatter on the same data source as LabelSet. When the scatter is commented out, the data doesn't update.

#### Complete, minimal, self-contained example code that reproduces the issue

```
# code from https://docs.bokeh.org/en/latest/docs/user_guide/annotations.html
# uncomment p.scatter line for correct behavior

from bokeh.plotting import figure, show, output_notebook
from bokeh.models import ColumnDataSource, Range1d, LabelSet, Label
output_notebook()

source = ColumnDataSource(data=dict(height=[],
                                    weight=[],
                                    names=[]))

p = figure(title='Dist. of 10th Grade Students at Lee High',
           x_range=Range1d(140, 275))
# p.scatter(x='weight', y='height', size=8, source=source)
p.xaxis[0].axis_label = 'Weight (lbs)'
p.yaxis[0].axis_label = 'Height (in)'

labels = LabelSet(x='weight', y='height', text='names', level='glyph',
              x_offset=5, y_offset=5, source=source, render_mode='canvas')

p.add_layout(labels)

h = show(p, notebook_handle=True)

source.data = {'height': [66, 71, 72, 68, 58, 62],
 'weight': [165, 189, 220, 141, 260, 174],
 'names': ['Mark', 'Amir', 'Matt', 'Greg', 'Owen', 'Juan']}
push_notebook(h)
```

**Comment by mattpap** [comment]
If you add an explicit `y_range`, e.g. `y_range=(60, 80)`, then updating works, because bokehjs can figure out the coordinate space and render labels accordingly. Without explicit `{x,y}_range` and without a glyph, there is no way do so and this results in an empty canvas. Annotations don't take part in such computations, contrary to glyphs, because annotations can be attached outside of the coordinate space. The preferred way of attaching labels to glyphs is to use `Text` glyph.

Bokeh already warns about not having any glyph renderers, which is good indicator that things may not work. Additionally bokehjs warns about not being able to compute initial ranges, though it does this in JS console, which usually isn't visible to the user.  We could make this warning more prominent to user, to make him understand more what's going on (barring technical difficulties with server setup, etc.).

I will leave this open as a discussion for the time being.

**Comment by mattpap** [comment]
Also to note, this issue isn't specific to the notebook. Things will fail similarly when using bokeh server.

**Comment by bryevdv** [comment]
> The preferred way of attaching labels to glyphs is to use Text glyph.

I don't agree with this statement. The explicit purpose of `LabelSet` is attaching labels to glyphs. The issue here is that there is no glyph.

**Comment by currivan** [comment]
> Bokeh already warns about not having any glyph renderers, which is good indicator that things may not work.

This warning is somehow suppressed in my working notebook, although I get it in a clean one. I haven't been able to isolate why.

**Comment by bryevdv** [comment]
> This warning is somehow suppressed in my working notebook, although I get it in a clean one. I haven't been able to isolate why.

The notebook does all kinds of thing, and does what it does. If it's not showing up then we may not be able to make it show up.
